src/event/writer.rs

```diff
@@ -112,37 +116,51 @@ impl EventWriter {
     /// ```ignore
     /// let mut event_writer = client_factory.create_event_writer(stream);
     /// // result is a tokio oneshot
-    /// let result = event_writer.write_event(payload).await;
-    /// result.await.expect("flush to server");
+    /// event_writer.write_event(payload).await;
+    /// event_writer.flush().await.expect("flush to server");
     /// ```
-    pub async fn write_event(&mut self, event: Vec<u8>) -> oneshot::Receiver<Result<(), Error>> {
+    pub async fn write_event(&mut self, event: Vec<u8>) {
```

Comment:
I don't think we can make this change.
Unlike a byteStream where if one write fails all those after it are sure to fail, and we can resume knowing the offset;
EventStreams have multiple segments, so one event can error while other events before and after can succeed. Similarly the order of acks may be different for different segments.

We can have a flush method, but it does not remove the need for the ack callbacks.